File: 1 - Frontend/src/app/core/domain/entity/consulta-model.ts
import { DomainModel } from './base/domain-model';
import { PessoaModel } from './pessoa-model';

export class ConsultaModel extends DomainModel {
  dataHoraInicio?: Date = null;
  dataHoraFinal?: Date = null;
  observacoes?: string = null;
  paciente?: PessoaModel = null;
}

File: 2 - Backend/AM.Application/ViewModel/ConsultaModel.cs
using System;

namespace AM.App.ViewModel
{
    public class ConsultaModel
    {
        public int Id { get; set; }
        public DateTime DataHoraInicio { get; set; }
        public DateTime DataHoraFinal { get; set; }
        public string Observacoes { get; set; }
        public int IdPessoa { get; set; }
        public PessoaModel Pessoa { get; set; }
    }
}
